lib/src/facts/freebsd/processor_resolver.cc

```diff
@@ -0,0 +1,50 @@
+#include <internal/facts/freebsd/processor_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace freebsd {
+
+    processor_resolver::data processor_resolver::collect_data(collection& facts)
+    {
+        auto result = posix::processor_resolver::collect_data(facts);
+        size_t len;
+        int mib[2];
+        mib[0] = CTL_HW;
+
+        // Get the logical count of processors
+        len = sizeof(result.logical_count);
+        mib[1] = HW_NCPU;
+
+        if (sysctl(mib, 2, &result.logical_count, &len, nullptr, 0) == -1) {
+            LOG_DEBUG("sysctl hw.ncpu failed: %1% (%2%): logical processor count is unavailable.", strerror(errno), errno);
+        }
+
+        // For each logical processor, collect the model name
+        if (result.logical_count > 0) {
+            // Note: we're using the same description string for all logical processors:
+            // using different CPUs is not even likely to work.
+            vector<char> buffer(256);
+
+            while (true) {
+                size_t size = buffer.size();
+                mib[1] = HW_MODEL;
+                if (sysctl(mib, 2, buffer.data(), &size, nullptr, 0) == 0) {
+                    buffer.resize(size + 1);
+                    result.models.resize(result.logical_count, buffer.data());
+                    break;
+                }
+                if (errno != ENOMEM) {
+                    LOG_DEBUG("sysctl hw.model failed: %1% (%2%): processor models are unavailable.", strerror(errno), errno);
+                    break;
+                }
+                buffer.resize(buffer.size() * 2);
+            }
+        }
```

Comment:
The FreeBSD doesn't seem to have 1:1 mapping with OpenBSD, the only constants(Are the all caps references to sysctl stuff called constants) I can find in the docs are:

```
           HW_MACHINE		      string	    no
	   HW_MODEL		      string	    no
	   HW_NCPU		      integer	    no
	   HW_BYTEORDER		      integer	    no
	   HW_PHYSMEM		      integer	    no
	   HW_USERMEM		      integer	    no
	   HW_PAGESIZE		      integer	    no
	   HW_FLOATINGPT	      integer	    no
	   HW_MACHINE_ARCH	      string	    no
	   HW_REALMEM		      integer	    no
```

For other settings, all the references and code I can find use the kenv lookup:

```
bios_release_date = kenv -q smbios.bios.reldate
bios_vendor = kenv -q smbios.bios.vendor
bios_version = kenv -q smbios.bios.version
boardmanufacturer = kenv -q smbios.planar.maker
boardproductname = kenv -q smbios.planar.product
boardserialnumber = kenv -q smbios.planar.serial
serialnumber = kenv -q smbios.system.serial
productname = kenv -q smbios.system.product
manufacturer = kenv -q smbios.system.maker
uuid = kenv -q smbios.system.uuid
```

```
root@freebsd-10-facter3:~ # sysctl smbios.bios.reldate
sysctl: unknown oid 'smbios.bios.reldate': No such file or directory
root@freebsd-10-facter3:~ # kenv smbios.bios.reldate
12/01/2006
```